api/grpcserver/transaction_service.go

```diff
@@ -20,6 +21,14 @@ type TransactionService struct {
 	Network api.NetworkAPI // P2P Swarm
 	Mesh    api.TxAPI      // Mesh
 	Mempool *state.TxMempool
+	syncer  Syncer
+}
+
+// Syncer defines ways of interacting with the node syncer
+// business logic
+type Syncer interface {
```

Comment:
Following up on the thread from the previous PR (https://github.com/spacemeshos/go-spacemesh/pull/2114#discussion_r482355573), in principle I agree with your point about the "interface segregation principle," but I don't think it's worth defining a new interface here for a single method - especially since, as I pointed out, the object being passed in already implements both methods.

In any case we do this many times in our code already - if we think that should be changed, we should probably do that as a separate issue.

Reply:
Ok, I'll then reuse the interface.